File: api/utils/bcrypt.py
import bcrypt


def hash_password(password: str) -> str:
    password = password.encode("utf-8")
    salt = bcrypt.gensalt()
    hashed = bcrypt.hashpw(password, salt)
    return hashed.decode("utf-8")


def check_password(password: str, hashed_password: str) -> bool:
    password = password.encode("utf-8")
    hashed_password = hashed_password.encode("utf-8")
    return bcrypt.checkpw(password, hashed_password)

File: src/utils/bcrypt.ts
import bcrypt from "bcrypt";

const SALT_ROUNDS = 12;

export const hashPassword = async ({ password }: { password: string }) => {
  return await bcrypt.hash(password, SALT_ROUNDS);
};

export const comparePassword = async ({
  password,
  hashedPassword,
}: {
  password: string;
  hashedPassword: string;
}) => {
  return await bcrypt.compare(password, hashedPassword);
};
